ReplayKitExample/BroadcastExtension/ExampleReplayKitAudioCapturer.m

```diff
@@ -0,0 +1,174 @@
+        "  category: %@\n"
+        "  categoryOptions: %ld\n"
+        "  mode: %@\n"
+        "  sampleRate: %.2f\n"
+        "  IOBufferDuration: %f\n"
+        "  outputNumberOfChannels: %ld\n"
+        "  inputNumberOfChannels: %ld\n"
+        "  outputLatency: %f\n"
+        "  inputLatency: %f\n"
+        "  outputVolume: %f\n"
+        "}";
+        AVAudioSession *session = [AVAudioSession sharedInstance];
+        NSString *description = [NSString stringWithFormat:format,
+                                 session.category , (long)session.categoryOptions, session.mode,
+                                 session.sampleRate, session.IOBufferDuration,
+                                 session.outputNumberOfChannels, session.inputNumberOfChannels,
+                                 session.outputLatency, session.inputLatency, session.outputVolume];
+    NSLog(@"%@", description);
+
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW,(int64_t)(1 * NSEC_PER_SEC)),
+                   dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0),
+                   ^{
+                       [self printAVAudioSession];
+                   }
+    );
+}
+
++ (NSString *)description {
+    return @"ExampleReplayKitAudioCapturer";
+}
+
+#pragma mark - TVIAudioDeviceRenderer
+
+- (nullable TVIAudioFormat *)renderFormat {
+    return nil;
+}
+
+- (BOOL)initializeRenderer {
+    return NO;
+}
+
+- (BOOL)startRendering:(nonnull TVIAudioDeviceContext)context {
+    return NO;
+}
+
+- (BOOL)stopRendering {
+    return NO;
+}
+
+#pragma mark - TVIAudioDeviceCapturer
+
+- (nullable TVIAudioFormat *)captureFormat {
+    if (!_capturingFormat) {
+        /*
+         * Assume that the AVAudioSession has already been configured and started and that the values
+         * for sampleRate and IOBufferDuration are final.
+         */
+        _capturingFormat = [[self class] activeCapturingFormat];
+    }
+
+    return _capturingFormat;
+}
+
+- (BOOL)initializeCapturer {
+    return YES;
+}
+
+- (BOOL)startCapturing:(nonnull TVIAudioDeviceContext)context {
+    @synchronized (self) {
+        NSAssert(capturingContext == NULL, @"Should not have any capturing context.");
+        capturingContext = malloc(sizeof(ExampleAudioContext));
+        capturingContext->deviceContext = context;
+        capturingContext->maxFramesPerBuffer = _capturingFormat.framesPerBuffer;
+
+        const NSTimeInterval sessionBufferDuration = [AVAudioSession sharedInstance].IOBufferDuration;
+        const double sessionSampleRate = [AVAudioSession sharedInstance].sampleRate;
+        const size_t sessionFramesPerBuffer = (size_t)(sessionSampleRate * sessionBufferDuration + .5);
+        capturingContext->expectedFramesPerBuffer = sessionFramesPerBuffer;
+
+        capturingContext->deviceContext = context;
+    }
+    return YES;
+}
+
+- (BOOL)stopCapturing {
+    @synchronized(self) {
+        capturingContext->deviceContext = NULL;
+    }
+
+    return YES;
+}
```

Comment:
I can see why you experienced a crash @piyushtank. It seems like we need to free `capturingContext` at this time.